### PR TITLE
B-15529: Support correlation by IP or MAC address

### PR DESCRIPTION
**Pull request by travisgoldie**
Sample JSON:

```json
{
    "environment": {
        "statusCode": "OK",
        "id": 250
    },
    "events": [
        {
            "layer": "Compute",
            "time": "2017-03-03T06:00:00-0456",
            "type": "SDK Event",
            "isNoise": true,
            "assocResourceCollection": {
                "cloudVMs": [
                    {
                        "networkInterfaces": [],
                        "uuid": "421bac04-f3d7-a600-ef5c-b3219e90d1eb"
                    },
                    {
                        "networkInterfaces": [
                            {
                                "networkAddresses": [],
                                "hwAddress": "00:50:56:9b:7c:31"
                            },
                            {
                                "networkAddresses": [],
                                "hwAddress": "00:50:56:9b:06:1c"
                            }
                        ],
                        "uuid": ""
                    },
                    {
                        "networkInterfaces": [
                            {
                                "networkAddresses": [
                                    {
                                        "address": "fe80::250:56ff:fe9b:5945"
                                    },
                                    {
                                        "address": "10.17.109.123"
                                    }
                                ],
                                "hwAddress": ""
                            }
                        ],
                        "uuid": ""
                    },
                    {
                        "networkInterfaces": [
                            {
                                "networkAddresses": [
                                    {
                                        "address": "fe80::250:56ff:fe93:c8eb"
                                    },
                                    {
                                        "address": "2001:5c0:110e:3368:250:56ff:fe93:c8eb"
                                    },
                                    {
                                        "address": "172.17.107.153"
                                    }
                                ],
                                "hwAddress": "00:50:56:93:c8:eb"
                            }
                        ],
                        "uuid": ""
                    }
                ]
            },
            "name": "SDK Event",
            "rootCauseResourceCollection": {
                "cloudVMs": []
            },
            "cloudEnvironment": {
                "healthState": "OK",
                "id": 250
            },
            "impactedResourceCollection": {
                "cloudVMs": []
            },
            "description": "Caused by phase of the moon.",
            "category": "Performance",
            "internalType": "ProviderThirdParty",
            "severity": "Info"
        }
    ]
}
```

### CODE REVIEW

File: python/test/scripts/example.py [hunk 1]
```diff
@@ -45,7 +47,36 @@ def main(args):
...
+        ),
+        CloudVM(
+            network_interfaces = [
+                NetworkInterface(ip_addresses = [
+                    "fe80::250:56ff:fe9b:5945", "10.17.109.123",
+                ]),
+            ],
+        ),
+        CloudVM(
+            network_interfaces = [
```
Comment (srazin) [inline python/test/scripts/example.py]: please maintain the property (while in python standard way) closer to the our API model to avoid the disconnect.

**Comment by srazin** [comment]
Travis is this PR ready for review or you currently still working through it?

**Comment by travisgoldie** [comment]
Waiting for verification with @thorste. We want to make sure that this code works for the use case.

**Comment by srazin** [comment]
ok. Poke once you ready.

-----
Regards,
Sergey A. Razin, Ph.D. |  Chief Technology Officer
SIOS Technology Corp.
P: 650.669.8914
E: sergey.razin@us.sios.com
LinkedIn: http://www.linkedin.com/in/sergeyrazin
Twitter @ TechDozor <https://twitter.com/techdozor>
www.us.sios.com

**Comment by travisgoldie** [comment]
POKE. Looks done.

**Review by srazin — APPROVED** [review]
looks good

**Comment by travisgoldie** [comment]
@srazin, i am not authorized to merge. Please merge.

**Review by srazin — APPROVED** [review]
test

**Pull request merged**